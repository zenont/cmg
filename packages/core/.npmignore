package-lock.json
tsconfig.*.json
tsconfig.*
rollup.*.js
node_modules