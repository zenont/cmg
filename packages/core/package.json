{
    "name": "@365-widgets/core",
    "version": "0.0.0",
    "main": "main/index.js",
    "module": "module/index.js",
    "sideEffects": false,
    "types": "typings/index.d.ts",
    "scripts": {
      "build": "rollup -c rollup.config.js",
      "watch": "rollup -c rollup.config.js --watch"
    }
  }
  