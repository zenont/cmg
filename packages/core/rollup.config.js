import babel from '@rollup/plugin-babel'
import commonjs from '@rollup/plugin-commonjs'
import resolve from '@rollup/plugin-node-resolve'

const extensions = ['.js', '.jsx', '.ts', '.tsx', '.mjs', '.json']
const sourcemap = 'inline'

export default {
  input: 'src/index.ts',
  output: [
    {
      file: 'main/index.js',
      sourcemap,
      format: 'cjs',
    },
    {
      file: 'module/index.js',
      sourcemap,
      format: 'esm',
    },
  ],
  plugins: [
    babel({
      exclude: 'node_modules/**',
      rootMode: 'upward',
      extensions,
    }),
    resolve({ extensions, resolveOnly: [/src/] }),
    commonjs({
      extensions,
    }),
  ],
}
