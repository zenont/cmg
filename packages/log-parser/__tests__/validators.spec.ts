import { SensorType } from '@365-widgets/core'
import {
  validateHumiditySensor,
  ReferenceResult,
  validateThermometer,
} from '../src'

describe('log-parser', () => {
  describe('evaluator', () => {
    it('validates the temperature sensor and is "precise"', async () => {
      const reference: ReferenceResult = {
        temperature: 70,
        humidity: 45,
        ppm: 6,
      }

      const result = validateThermometer(reference, {
        sensorType: SensorType.Thermometer,
        min: 61.4,
        max: 79.1,
        sumValue: 905.2,
        totalCount: 13,
      })

      expect(result).toMatchSnapshot()
    })

    it('validates the temperature sensor and is "very precise"', async () => {
      const reference: ReferenceResult = {
        temperature: 70,
        humidity: 45,
        ppm: 6,
      }

      const result = validateThermometer(reference, {
        sensorType: SensorType.Thermometer,
        min: 69.5,
        max: 71.5,
        sumValue: 360,
        totalCount: 5,
      })

      expect(result).toMatchSnapshot()
    })

    it('validates the temperature sensor and is "ultra precise"', async () => {
      const reference: ReferenceResult = {
        temperature: 70,
        humidity: 45,
        ppm: 6,
      }

      const result = validateThermometer(reference, {
        sensorType: SensorType.Thermometer,
        min: 69.5,
        max: 71.5,
        sumValue: 352.2,
        totalCount: 5,
      })

      expect(result).toMatchSnapshot()
    })

    it('validates the humidity sensor and is "keep"', async () => {
      const reference: ReferenceResult = {
        temperature: 70,
        humidity: 45,
        ppm: 6,
      }

      const result = validateHumiditySensor(reference, {
        sensorType: SensorType.Humidity,
        min: 45.1,
        max: 45.3,
        sumValue: 135.6,
        totalCount: 3,
      })

      expect(result).toMatchSnapshot()
    })

    it('validates the humidity sensor and is "discard"', async () => {
      const reference: ReferenceResult = {
        temperature: 70,
        humidity: 45,
        ppm: 6,
      }

      const result = validateHumiditySensor(reference, {
        sensorType: SensorType.Humidity,
        min: 42.1,
        max: 44.9,
        sumValue: 219.1,
        totalCount: 5,
      })

      expect(result).toMatchSnapshot()
    })
  })
})
