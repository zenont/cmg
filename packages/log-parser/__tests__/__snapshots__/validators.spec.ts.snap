// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`log-parser evaluator validates the humidity sensor and is "discard" 1`] = `"discard"`;

exports[`log-parser evaluator validates the humidity sensor and is "keep" 1`] = `"keep"`;

exports[`log-parser evaluator validates the temperature sensor and is "precise" 1`] = `"precise"`;

exports[`log-parser evaluator validates the temperature sensor and is "ultra precise" 1`] = `"ultra precise"`;

exports[`log-parser evaluator validates the temperature sensor and is "very precise" 1`] = `"precise"`;
